lib/src/fsmonitor.rs

```diff
@@ -44,20 +48,27 @@ pub enum FsmonitorKind {
     None,
 }
 
-impl FromStr for FsmonitorKind {
-    type Err = config::ConfigError;
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s {
-            "watchman" => Ok(Self::Watchman),
-            "test" => Err(config::ConfigError::Message(
-                "cannot use test fsmonitor in real repository".to_string(),
-            )),
-            "none" => Ok(Self::None),
-            other => Err(config::ConfigError::Message(format!(
-                "unknown fsmonitor kind: {}",
-                other
-            ))),
+impl FsmonitorKind {
```

Comment:
comment (non-blocking): Since it's no longer a simple enum, we might like to rename it to `FsmonitorConfig`/`FsmonitorSettings`/`FsmonitorInfo` or something like that. I think not a big deal to do in this PR.

Reply:
`FsmonitorSettings`